drivers/ethernet/eth_adin2111.c

```diff
@@ -758,10 +891,14 @@ static int adin2111_await_device(const struct device *dev)
 
 	/* await reset complete (RESETC) and clear it */
 	for (count = 0U; count < ADIN2111_RESETC_AWAIT_RETRY_COUNT; ++count) {
-		ret = eth_adin2111_reg_read(dev, ADIN2111_STATUS0, &val);
+		ret = eth_adin2111_reg_read(dev, ADIN2111_PHYID, &val);
```

Comment:
Not sure about that change. The RM mentions STATUS0 poll. Will need to test.

Do you use HW reset? If no, then it could be related to clock stabilization delay (_done within hw reset_).

Reply:
Thaks, 

i tested mainly using the SW reset. Anyway i see that keeping the reset button pressed adin2111 clock stops.

Reading a read-only reg currently may not be the best, sure, but it works reliably, and allows "Generic SPI without CRC" to work.

What about if i move the check in the phy code ?

Anyway, asking also suggestions to the adi team tight now.



I see that

Reply:
@spectrum70 in cases when hw reset is not used does increasing the delay from 10ms to `ADIN2111_HW_BOOT_DELAY_MS` helps?

Reply:
> Anyway, i am asking also suggestions to the adi team right now.

Was there any response from them?